docs/timeseries_pipeline_elements.md

```diff
@@ -0,0 +1,367 @@
+# Timeseries Pipelines [<sup><mark>experimental</mark></sup>](/docs/README.md#tag-notes)
+
+> [Description](#timeseries-pipeline-description)<br>
+> [Example](#timeseries-pipeline-example)<br>
+> [Pipeline Elements](#timeseries-pipeline-elements)
+
+## Description <a id="timeseries-pipeline-description"></a>
+
+Timescale timeseries objects are just a convenient and efficient way of tracking a single value over time and are detailed a bit more [here](timeseries.md).  One of our primary goals with timeseries is that they should be easy and efficient to perform basic operations on, and that is where pipelines enter the picture.  At its simplest, a pipeline is just a timeseries connected to a [pipeline element](#timeseries-pipeline-elements) via the pipeline operator `|>`.  However, most pipeline operations output new timeseries, so it's possible to chain many pipeline elements together such that the output from one element become the input to the next.
+
+### A note on operator associativity and grouping
+
+Due to limitations in the PostgresQL parser, custom operators are required to be left associative.  The following pipeline will always result in `elementA` being applied to `timeseries` and then `elementB` being applied to the result.
+
+```SQL ,ignore
+SELECT timeseries |> elementA |> elementB;
+```
+
+However, it is possible to explicitly group elements using parentheses:
+
+```SQL ,ignore
+SELECT timeseries |> (elementA |> elementB);
+```
+
+This will result in a pipeline object being created from elements A and B, which will then be applied to the timeseries.  While we don't presently take maximum advantage of this internally, these multiple element pipelines should enable optimizations moving forward.  Therefore, this second form should be preferred where possible.
```

Comment:
"Therefore, this second form should be preferred where possible"

This feels like guidance that's just way too general. Are there drawbacks? Is there a time you wouldn't want to do that? IDK, just not sure where to go with this as a user because I'm sure at some point it won't always work as I want??

Anyway, just some feedback to consider.